components/bl-category-chart/src/index.js

```diff
@@ -43,104 +17,108 @@ export default function CategoryChartComponent({ component, elRef }) {
       ref={ elRef }
       style={ styles }
       className={
-        cn("bl-customComponent-categoryChart", classList, { "bl-customComponent-categoryChart--disabled": disabled })
+        cn('bl-customComponent-categoryChart', classList, { 'bl-customComponent-categoryChart--disabled': disabled })
       }>
-      <canvas ref={ chartRef } style={{ backgroundColor: chartBackgroundColor, width: "100%", height: "100%" }} />
+      <canvas ref={ chartRef } style={{ backgroundColor, width: '100%', height: '100%' }}/>
     </div>
   );
 }
 
-const useChart = (chartRef, props) => {
+function useChart(component) {
   const {
-    display, type, labels, datasets, options, chartTitleVisibility, chartTitleFontSize,
-    chartTitle, yGridLineVisibility, xGridLineVisibility, gridLinesColor, gridLinesWidth
-  } = props;
-  const chartInstance = useRef();
+    options, yGridLineVisibility, xGridLineVisibility, gridLinesColor, gridLinesWidth,
+    legendVisibility, titleVisibility, title, titleFontSize,
+    display, type, labels, datasets
+  } = component;
+
+  const chartRef = useRef(null);
+  const [chartInstance, setChartInstance] = useState(null);
+
+  const chartData = useMemo(() => ({ labels, datasets }), [labels, datasets]);
+
+  const pluginsOptions = useMemo(() => ({
+    legend: { display: legendVisibility },
+    title: {
+      display: titleVisibility,
+      text: title,
+      font: {
+        size: titleFontSize,
+      },
+    },
+  }), [legendVisibility, titleVisibility, title, titleFontSize]);
+
+  const scaleOptions = useMemo(() => ({
+    y: {
+      grid: {
+        display: yGridLineVisibility,
+        color: gridLinesColor,
+        lineWidth: gridLinesWidth,
+      },
+    },
+    x: {
+      grid: {
+        display: xGridLineVisibility,
+        color: gridLinesColor,
+        lineWidth: gridLinesWidth,
+      },
+    },
+  }), [yGridLineVisibility, xGridLineVisibility, gridLinesColor, gridLinesWidth]);
 
   useEffect(() => {
     if (!display) {
-      return null;
+      return;
     }
```

Comment:
Do we really need it?

I assume this useEffect shouldn't rely on display

Reply:
We need this dependency because without it, we get a context getting error when creating the chart.